hippy/module/pypy_bridge/py_strategies.py

```diff
@@ -97,6 +97,33 @@ def pop(self, w_list, index):
 
         return w_php_val.to_py(interp)
 
+    def getslice(self, w_list, start, stop, step, length):
+        """Adapted from the PyPy implementation"""
+
+        interp = self.space.get_php_interp()
+        w_php_arry_ref = self.unerase(w_list.lstorage)
+        w_php_arry = w_php_arry_ref.deref_temp()
+
```

Comment:
The problem with ``.lst_w`` btw, is that ``w_php_array`` is not known to be a ``W_ListArrayObject`` at this point (the annotator thinks its a ``W_Root``. so you need to add an ``assert isinstance(w_php_array)`` here, which I think is better than adding a ``W_ListArrayObject._get_raw_storage`` method that I don't understand.